bigquery/gcloud/aio/bigquery/table.py

```diff
@@ -0,0 +1,315 @@
+        project = await self.project()
+        url = f'{API_ROOT}/projects/{project}/jobs'
+
+        body = self._make_copy_body(
+            project, destination_project,
+            destination_dataset, destination_table)
+        response = await self._post_json(url, body, session, timeout)
+        return Job(response['jobReference']['jobId'], self._project,
+                   session=self.session, token=self.token)
+
+    # https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/insert
+    # https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfigurationLoad
+    async def insert_via_load(
+            self, source_uris: List[str], session: Optional[Session] = None,
+            autodetect: bool = False,
+            source_format: SourceFormat = SourceFormat.CSV,
+            write_disposition: Disposition = Disposition.WRITE_TRUNCATE,
+            timeout: int = 60,
+            ignore_unknown_values: bool = False,
+            schema_update_options: Optional[List[SchemaUpdateOption]] = None
+    ) -> Job:
+        """Loads entities from storage to BigQuery."""
+        project = await self.project()
+        url = f'{API_ROOT}/projects/{project}/jobs'
+
+        body = self._make_load_body(
+            source_uris, project, autodetect, source_format, write_disposition,
+            ignore_unknown_values, schema_update_options or []
+        )
+        response = await self._post_json(url, body, session, timeout)
+        return Job(response['jobReference']['jobId'], self._project,
+                   session=self.session, token=self.token)
+
+    # https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/insert
+    # https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfigurationQuery
+    async def insert_via_query(
+            self, query: str, session: Optional[Session] = None,
+            write_disposition: Disposition = Disposition.WRITE_EMPTY,
+            timeout: int = 60, use_query_cache: bool = True,
+            dry_run: bool = False) -> Job:
+        """Create table as a result of the query"""
+        warnings.warn('using Table#insert_via_query is deprecated.'
+                      'use Job#insert_via_query instead', DeprecationWarning)
+        project = await self.project()
+        url = f'{API_ROOT}/projects/{project}/jobs'
+
+        body = self._make_query_body(query, project, write_disposition,
+                                     use_query_cache, dry_run)
+        response = await self._post_json(url, body, session, timeout)
+        job_id = response['jobReference']['jobId'] if not dry_run else None
+        return Job(job_id, self._project,
+                   session=self.session, token=self.token)
+
+    async def close(self) -> None:
+        await self.session.close()
+
+    async def __aenter__(self) -> 'Table':
+        return self
+
+    async def __aexit__(self, *args: Any) -> None:
+        await self.close()
```

Comment:
Please move these three methods to the new base class -- we want them to be defined in the same place we're describing what `self.session` is and will want them inherited into the new Dataset class.

Reply:
done.